ion-schema/src/model/constraints/any_of.rs

```diff
@@ -0,0 +1,174 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::internal_traits::{
+    LoaderContext, ValidateInternal, ValidationContext, WriteAsIsl, WriteContext,
+};
+use crate::ion_schema_version::Versioned;
+use crate::model::constraints::{ConstraintName, ReadConstraint};
+use crate::model::type_argument::TypeArgument;
+use crate::model::{TypeDefinitionBuilder, VersionedTypeArgumentList};
+use crate::result::IonSchemaResult;
+use crate::{IonSchemaElement, IslVersion, ViolationRecorder};
+use ion_rs::{Element, ValueWriter};
+use std::ops::ControlFlow;
+
+impl ConstraintName for AnyOf {
+    const CONSTRAINT_NAME: &'static str = "any_of";
+}
+
+/// Represents the `any_of` constraint (ref. [ISL 1.0], [ISL 2.0]).
+///
+/// [ISL 1.0]: https://amazon-ion.github.io/ion-schema/docs/isl-1-0/spec#any_of
+/// [ISL 2.0]: https://amazon-ion.github.io/ion-schema/docs/isl-2-0/spec#any_of
+#[derive(Debug, PartialEq, Clone)]
+pub struct AnyOf {
+    type_arguments: Vec<TypeArgument>,
+}
+
+impl AnyOf {
+    fn new(mut type_arguments: Vec<TypeArgument>) -> Self {
+        // Sorting the vec allows us to get Bag-like equality semantics.
+        // TODO: Replace the vec with a HashBag or come up with a less hacky way to sort the type arguments.
```

Comment:
Since this is used in a few places, consider making a naive `Bag` type wrapping a sorted `Vec<TypeArgument>` that you can optimize later.